parso/tree.py

```diff
@@ -1,22 +1,23 @@
+import warnings
 from abc import abstractmethod, abstractproperty
-from typing import List, Optional, Tuple
+from typing import List, Optional, Tuple, Union
 
 from parso.utils import split_lines
 
 
-def search_ancestor(node, *node_types):
+def search_ancestor(node: 'NodeOrLeaf', *node_types: str) -> 'Optional[BaseNode]':
     """
     Recursively looks at the parents of a node and returns the first found node
-    that matches node_types. Returns ``None`` if no matching node is found.
+    that matches ``node_types``. Returns ``None`` if no matching node is found.
+
+    This function is deprecated, use :meth:`NodeOrLeaf.search_ancestor` instead.
 
     :param node: The ancestors of this node will be checked.
     :param node_types: type names that are searched for.
-    :type node_types: tuple of str
     """
-    while True:
-        node = node.parent
-        if node is None or node.type in node_types:
-            return node
+    warnings.warn("parso.tree.search_ancestor() is deprecated, "
+                  "use NodeOrLeaf.search_ancestor() instead", DeprecationWarning)
```

Comment:
Can we maybe not have this warning yet? I know that a lot of people will be annoyed by it in IPython (because of some logger shenanigans) once we release the next parso version.

In general I feel like this is a good thing, but we should probably wait until Jedi uses the newer version. IMO there's no reason to hurry.

Reply:
Thanks!